Source/Core/VideoBackends/D3D12/Render.cpp

```diff
@@ -1244,6 +1242,8 @@ void Renderer::SetSamplerState(int stage, int tex_index, bool custom_tex)
 	if (tex_index)
 		stage += 4;
 
+	const auto original_hex = gx_state.sampler[stage].hex;
```

Comment:
That was a mistake in communication on my part (I misread the `.hex as belonging to something in BPMemory, of which some use union and C bitfield combinations (I plan to correct this once the feature freeze is lifted)).

(feel free to skip below if you don't care about language-lawyer-esque stuff, I'd understand)

As an aside, with regards to BitField in our Common library, I (arguably) think it shouldn't be used at all, considering providing an actual API for accessing values from the central primitive in a struct/class/whatever is better in the long-run, since it enforces intent better and models how your codebase intends to use the given dataset better (in many cases, all that's wanted is to read a value from a specific set of bits, but not store to them), and has more desirable traits (no cast required for printf/stream output; .Value() can be called on BitField - though annoying, no reliance on class/struct packing in order to function correctly, no reliance on forced inlining, etc).

They're also clunky when used with a union since some bitfield entries are used as boolean values to dictate when to do certain behaviors. However, in order to not be undefined behavior, accessing of a union member that isn't active can only be done if it's part of the common initial sequence of the union (all members having the same type). This makes enforcing type correctness kind of a pain in the neck, and can be a cause of subtle bugs that the compiler would have otherwise flagged down with a warning.